lib/jarvis/travis/watchdog.rb

```diff
@@ -0,0 +1,76 @@
+require "open-uri"
+require "travis"
+require "yaml"
+
+module Jarvis module Travis
+  class Watchdog
+    DEFAULT_PLUGINS_URL = "https://raw.githubusercontent.com/elastic/logstash/master/rakelib/plugins-metadata.json"
+    PLUGINS_CONFIG_FILE = File.join(File.dirname(__FILE__), "../../../plugins_config.yml")
+    PLUGINS_CONFIG = YAML.load(File.read(PLUGINS_CONFIG_FILE))
+    DEFAULT_BRANCHES = ["master"]
+
+    def initialize
+    end
+
+    def default_plugins
+      @default_plugins ||= fetch(DEFAULT_PLUGINS_URL).select { |_, v| v["default-plugins"] }.keys
+    end
+
+    def get_status
+      status = {}
+
+      default_plugins.each do |plugin_name|
+        location = "logstash-plugins/#{plugin_name}"
+        repo = ::Travis::Repository.find(location)
+        branches = branch_to_monitor(plugin_name)
+        status[plugin_name] = branches.each_with_object({}) { |branch, hsh| hsh[branch] = repo.branches[branch].passed?  }
```

Comment:
When you use `passed?` at the *build* level it will ignore jobs marked as allowed to fails (like for the feature/9000), But you can check status per job.